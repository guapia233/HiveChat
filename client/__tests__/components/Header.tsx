import { render, screen } from "@testing-library/react";
import Header from "@/components/Header";
import "@testing-library/jest-dom";

describe("Header", () => {
  test("renders the page header", () => {
    // Render
    const { asFragment } = render(<Header />);

    // Sanity check
    expect(screen.getByText("Source code")).toBeInTheDocument();

    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });
});
