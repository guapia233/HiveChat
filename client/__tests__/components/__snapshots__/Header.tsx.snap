// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the page header 1`] = `
<DocumentFragment>
  <div
    class="flex m-3"
  >
    <img
      alt="Boost logo"
      data-nimg="1"
      decoding="async"
      height="60"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=128&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=64&q=75 1x, /_next/image?url=%2Fimg.jpg&w=128&q=75 2x"
      style="color: transparent;"
      width="60"
    />
    <div
      class="flex-1 flex justify-end align-middle"
    >
      <div
        class="flex flex-col justify-center pr-12 pl-12"
      >
        <a
          class="no-underline text-2xl text-black"
          href="https://github.com/anarthal/servertech-chat"
        >
          Source code
        </a>
      </div>
      <div
        class="flex flex-col justify-center pr-12 pl-12"
      >
        <a
          class="no-underline text-2xl text-black"
          href="https://anarthal.github.io/servertech-chat/"
        >
          Docs
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
